Models/GlamwireDb.cs

```diff
@@ -0,0 +1,160 @@
+    string connectionString = "Data Source=(localdb)\\MSSQLLocalDB;Initial Catalog=GlamwireDb;Integrated Security=True;Connect Timeout=30;Encrypt=False;Trust Server Certificate=False;Application Intent=ReadWrite;Multi Subnet Failover=False";
+
+    /// <summary>
+    /// initializes a new instance of the GlamwireDb, 
+    /// retrieves data from the database and populates it into the application
+    /// </summary>
+    public GlamwireDb()
+    {
+        try
+        {
+            // Code to connect to the database and retrieve data goes here
+            using SqlConnection connection = new SqlConnection(connectionString);
+        }
+        catch (Exception ex)
+        {
+            // Handle exceptions related to database connection and data retrieval
+            Console.WriteLine($"An error occurred while connecting to the database: {ex.Message}");
+        }
+    }
+
+
+    /// <summary>
+    /// Retrieve all NPCs from the database
+    /// </summary>
+    /// <param name="npcId"></param>
+    /// <returns></returns>
+    public List<NPC> RetrieveAllNPCs(int npcId)
+    {
+        List<NPC> npcs = new List<NPC>();
+
+        // placeholder to retrieve all NPCs from the database to populate and move them.
+        // // "hey program, try using the connection string to connect to the database"
+        try
+        {
+            using SqlConnection connection = new SqlConnection(connectionString);
+            // open the connection (a bridge to the database)
+            connection.Open();
+
+            // create the query to reference the NPC table, just for readability. 
+            string query = @"SELECT * FROM NPC";
+            using SqlCommand cmd = new SqlCommand(query, connection);
+            // use the reader to execute the command and read it from the database.
+            using SqlDataReader reader = cmd.ExecuteReader();
+
+            // use a while loop to read each row of data. 
+            // while there is data to read, keep reading it. 
+            // if not, exit the loop.
+            while (reader.Read())
+            {
+                // adding a new npc object to the npc list. 
+                npcs.Add(new NPC {
+                    // go through each column of the NPC table
+                    // and read the data (if any) into the npc object properties.
+                    NPCId = npcId,
+                    NPCFirstName = Convert.ToString(reader["NPCFirstName"]),
+                    NPCLastName = Convert.ToString(reader["NPCLastName"]),
+                    NPCUsername = Convert.ToString(reader["NPCUsername"]),
+                    NPCRole = Convert.ToString(reader["NPCRole"]),
+                    PersonalityType = Convert.ToString(reader["PersonalityType"]),
+                    IsLocked = Convert.ToBoolean(reader["IsLocked"]),
+                    IsGuilty = Convert.ToBoolean(reader["isGuilty"]),
+                });
+            }
+        }
+
+        catch (Exception ex) {
```

Comment:
I would recommend catching a `SqlException` so we know the problem is database related